core/domain/story_services.py

```diff
@@ -0,0 +1,318 @@
+# Copyright 2018 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Commands that can be used to operate on stories.
+
+All functions here should be agnostic of how StoryModel objects are
+stored in the database. In particular, the various query methods should
+delegate to the Story model class. This will enable the story
+storage model to be changed without affecting this module and others above it.
+"""
+
+import copy
+
+from core.domain import story_domain
+from core.platform import models
+import feconf
+
+(story_models,) = models.Registry.import_models([models.NAMES.story])
+datastore_services = models.Registry.import_datastore_services()
+memcache_services = models.Registry.import_memcache_services()
+
+# This takes additional 'title' parameters.
+CMD_CREATE_NEW = 'create_new'
+
+# Name for the story search index.
+SEARCH_INDEX_STORIES = 'stories'
+
+# The maximum number of iterations allowed for populating the results of a
+# search query.
+MAX_ITERATIONS = 10
+
+
+def _migrate_story_contents_to_latest_schema(versioned_story_contents):
+    """Holds the responsibility of performing a step-by-step, sequential update
+    of the story structure based on the schema version of the input
+    story dictionary. If any of the current story schemas
+    change, a new conversion function must be added and some code appended to
+    this function to account for that new version.
+
+    Args:
+        versioned_story_contents: A dict with two keys:
+          - schema_version: str. The schema version for the story.
```

Comment:
There's still some confusion between the schema version for the story contents and the schema version for the story; see above.

Reply:
Done.